server/pages/index.js

```diff
@@ -318,13 +316,62 @@ function setUpLoggedInRoute( req, res, next ) {
 			res.redirect( redirectUrl );
 			return;
 		}
+
+		const user = require( 'user-bootstrap' );
+
 		start = new Date().getTime();
 
 		debug( 'Issuing API call to fetch user object' );
-		user( req.cookies.wordpress_logged_in, geoCountry, function( error, data ) {
-			let searchParam, errorMessage;
 
-			if ( error ) {
+		user( req.cookies.wordpress_logged_in, geoCountry )
+			.then( data => {
+				let searchParam;
+
+				const end = new Date().getTime() - start;
+
+				debug( 'Rendering with bootstrapped user object. Fetched in %d ms', end );
+				req.context.user = data;
+
+				// Setting user in the state is safe as long as we don't cache it
+				setCurrentUserOnReduxStore( data, req.context.store );
+
+				if ( data.localeSlug ) {
+					req.context.lang = data.localeSlug;
+					req.context.store.dispatch( {
+						type: LOCALE_SET,
+						localeSlug: data.localeSlug,
+						localeVariant: data.localeVariant,
+					} );
+				}
+
+				if ( req.path === '/' && req.query ) {
+					searchParam = req.query.s || req.query.q;
```

Comment:
I think you can move the variable declaration from [L328](https://github.com/Automattic/wp-calypso/pull/26973/files#diff-b2d6784517d1151d7e75f0287d0c6478R328) here (and probably use `const`).